src/patient-search/patient-search.component.tsx

```diff
@@ -1,10 +1,96 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { match } from "react-router";
+import { debounce } from "lodash";
+import styles from "./patient-search.component.css";
+import { performPatientSearch } from "./patient-search.resource";
+import PatientSearchResults from "./patient-search-result/patient-search-result.component";
 
 export default function PatientSearch(props: PatientSearchProps) {
-  return <section>The patient search!</section>;
+  const searchTimeout = 400;
+  const [searchResults, setSearchResults] = useState([]);
+  const [emptyResult, setEmptyResult] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+
+  useEffect(() => {
+    if (searchTerm) {
+      performPatientSearch(searchTerm, "full").then(response => {
+        const { data } = response;
+        setSearchResults(data["results"]);
+        if (data["results"].length === 0) {
+          setEmptyResult(true);
+        } else {
+          setEmptyResult(false);
+        }
+      });
+    } else {
+      setSearchResults([]);
+    }
+  }, [searchTerm]);
+
+  const handleChange = debounce(searchTerm => {
+    setSearchTerm(searchTerm);
+  }, searchTimeout);
+
+  return (
+    <div>
+      <div className={styles.patientSearchHeader}>
+        <button onClick={props.history.goBack} className={styles.iconBtn}>
```

Comment:
Does the click handler here work? Unless react-router is doing something to auto-bind `goBack`, then the context to the function will be wrong and it won't work. Notice how assigning the `history.back` variable to a new variable results in a different behavior (because of javascript context) then if you call `history.back()` directly.

<img width="359" alt="image" src="https://user-images.githubusercontent.com/5524384/64586007-0ec6c080-d358-11e9-8ac8-ac28786564de.png">

Reply:
The click handler did work, not sure how, haven't been able to find anything online to explain why. However since there is no back arrow anymore so I removed it.